src/libraries/System.Runtime/tests/System/HalfTests.cs

```diff
@@ -0,0 +1,937 @@
+        }
+
+        [Theory]
+        [MemberData(nameof(Parse_Invalid_TestData))]
+        public static void Parse_Invalid(string value, NumberStyles style, IFormatProvider provider, Type exceptionType)
+        {
+            bool isDefaultProvider = provider == null || provider == NumberFormatInfo.CurrentInfo;
+            Half result;
+            if ((style & ~(NumberStyles.Float | NumberStyles.AllowThousands)) == 0 && style != NumberStyles.None && (style & NumberStyles.AllowLeadingWhite) == (style & NumberStyles.AllowTrailingWhite))
+            {
+                // Use Parse(string) or Parse(string, IFormatProvider)
+                if (isDefaultProvider)
+                {
+                    Assert.False(Half.TryParse(value, out result));
+                    Assert.Equal(default(Half), result);
+
+                    Assert.Throws(exceptionType, () => Half.Parse(value));
+                }
+
+                Assert.Throws(exceptionType, () => Half.Parse(value, provider: provider));
+            }
+
+            // Use Parse(string, NumberStyles, IFormatProvider)
+            Assert.False(Half.TryParse(value, style, provider, out result));
+            Assert.Equal(default(Half), result);
+
+            Assert.Throws(exceptionType, () => Half.Parse(value, style, provider));
+
+            if (isDefaultProvider)
+            {
+                // Use Parse(string, NumberStyles) or Parse(string, NumberStyles, IFormatProvider)
+                Assert.False(Half.TryParse(value, style, NumberFormatInfo.CurrentInfo, out result));
+                Assert.Equal(default(Half), result);
+
+                Assert.Throws(exceptionType, () => Half.Parse(value, style));
+                Assert.Throws(exceptionType, () => Half.Parse(value, style, NumberFormatInfo.CurrentInfo));
+            }
+        }
+
+        public static IEnumerable<object[]> ToString_TestData()
+        {
+            yield return new object[] { -4568.0f, "G", null, "-4568" };
+            yield return new object[] { 0.0f, "G", null, "0" };
+            yield return new object[] { 4568.0f, "G", null, "4568" };
+
+            yield return new object[] { float.NaN, "G", null, "NaN" };
+
+            //yield return new object[] { 2468.0f, "N", null, "2,468.00" };
+
+            // Changing the negative pattern doesn't do anything without also passing in a format string
+            var customNegativePattern = new NumberFormatInfo() { NumberNegativePattern = 0 };
+            yield return new object[] { -6312.0f, "G", customNegativePattern, "-6312" };
+
+            var customNegativeSignDecimalGroupSeparator = new NumberFormatInfo()
+            {
+                NegativeSign = "#",
+                NumberDecimalSeparator = "~",
+                NumberGroupSeparator = "*"
+            };
+            //yield return new object[] { -2468.0f, "N", customNegativeSignDecimalGroupSeparator, "#2*468~00" };
```

Comment:
some dead code?

Reply:
Nah, these had to be turned on. Fixed them now.